src/project.c

```diff
@@ -772,6 +772,36 @@ static gboolean update_config(const PropertyDialogElements *e, gboolean new_proj
 		}
 		g_free(locale_path);
 	}
+
+	/* create filename path if it doesn't exist and if file name ends with '.geany' */
+	if ((err_code = utils_is_file_writable(locale_filename)) != 0 &&
```

Comment:
Since the new code doesn't use the value assigned to `err_code` here, doesn't seem much point in assigning to it.

Moreover, this use of `utils_is_file_writable()` seems like it's deceptively subtle, given what that function does and the comment above it. It might also result in a weird race condition.

Since you're using `utils_mkdir(..., TRUE)` below, which will succeed if the directory already exists, I think you could probably just drop this whole `if` block. Just get the dirname of the project file and try and create it with `utils_mkdir(..., TRUE)` as you already do.

Given the above, the logic could probably be restructured like this (untested/example code):

```c
gchar *project_file_dirname = g_path_get_dirname(locale_filename);
gboolean create_dir = FALSE;

// if the project file's directory doesn't exist, offer the user to create it
if (! g_file_test(project_file_dirname, G_FILE_TEST_EXISTS))
    create_dir = dialogs_show_question_full(...);

// if the project file's directory doesn't exist and the user wants to try and create it
if (create_dir)
{
    // try and create the project file's directory recursively, if it doesn't already exist
    err_code = utils_mkdir(project_file_dirname, TRUE);
    if (err_code != 0)
    {
        SHOW_ERR1(...);
        g_free(project_file_dirname);
        return FALSE;
    }
}
else
{
    // TODO: should this be handled?
}

g_free(project_file_dirname);
```

Reply:
Thanks @codebrainz I applied this change.

I think my change was leaking `locale_filename`, so I did the same logic used for `base_path`:

```diff
+               if (err_code != 0)
                {
-                       g_free(filename_path);
+                       SHOW_ERR1(_("Filename path could not be created (%s)."), g_strerror(err_code));
+                       gtk_widget_grab_focus(e->file_name);
+                       utils_free_pointers(2, project_file_dirname, locale_filename, NULL);
+                       return FALSE;
                }
        }
```
I also think the TODO should be handled the same way as when the user opts for not creating the `Base Path`, so we have consistency:
```diff
+       else
+       {
-                // TODO: should this be handled?
+               gtk_widget_grab_focus(e->file_name);
+               utils_free_pointers(2, project_file_dirname, locale_filename, NULL);
+               return FALSE;
+       }
+
+       g_free(project_file_dirname);
```

Reply:
Ops. I introduced a bug in this else, when the path already exists.
I will fix here.

Reply:
> Ops. I introduced a bug in this else, when the path already exists.
> I will fix here.

Fixed!